lib/Cake/Model/Datasource/DboSource.php

```diff
@@ -857,7 +857,13 @@ public function name($data) {
  * @return bool True if the database is connected, else false
  */
 	public function isConnected() {
-		return $this->connected;
+	    try{
+            $connected = $this->_connection->query('Select 1');
+        }catch (Exception $e){
```

Comment:
The formatting here needs to match the coding standards for the rest of the project.